### PR TITLE
ENH: Exit `save()` faster, if clean (fixes gh-797)

### PR DESCRIPTION
**Pull request by mih**

### CODE REVIEW

File: datalad/interface/save.py [hunk 1]
```diff
@@ -108,7 +108,24 @@ class Save(Interface):
...
+                index=True,
+                working_tree=True,
+                untracked_files=True,
+                submodules=True):
+            # if we cannot see anything dirty at all, the only things we could
+            # do is tag
+            if version_tag:
+                ds.repo.tag(version_tag)
+            # take the easy one out
+            return False
```
Comment (yarikoptic) [inline datalad/interface/save.py]: Did it ever return true? If not better just to return None

**Comment by yarikoptic** [comment]
I will try to check with codecov wtf is happening with coverage - forbids me merging on the phone

File: datalad/interface/save.py (hunk 1 above)
Comment (mih) [inline datalad/interface/save.py]: OK, will require further changes though to be uniform.

**Comment by mih** [comment]
re coverage report: This is not specific to this PR.

**Comment by mih** [comment]
Rebased on master to get `add(save=False)`. RF'ed `save` to use top-level `add` to handle saving of files in a dataset.

**Comment by mih** [comment]
@bpoldrack please take a look at this failure. I don't see how this is supposed to work. Otherwise, I have to stop using `add` internally and switch back to repo.add()

**Comment by bpoldrack** [comment]
At a first glance, the failure raises because you try to add everything, that might be possible to add to annex and then everything left to git. Therefore you call `Add.__call__` two times, setting `to_git` to `False` and  `True` respectively. And `Add` raises, since there is no annex and you wanted to add something with `to_git=False`.

So, what do we want? That `Add` doesn't fail, but is just doing nothing or checking for an annex in `Save` before calling `Add.__call__( ... to_git=False)`? I'd prefer the latter.

**Comment by bpoldrack** [comment]
Actually, I'm wrong. `Add` should not raise, since this is troublesome with recursive calls, but skip these paths and return `success=False` for them. I'll send you a PR.

**Comment by mih** [comment]
We have a winner -> merge.

**Pull request merged**